### ISSUE TITLE
Calling of unmapped URI entails blending

### ISSUE THREAD

**Issue by un-tone** [Starcounter/Home#251]
I tested on WebsiteEditor app.

#### Steps to reproduce

1. Run **WebsiteEditor** and **WebsiteProvider** apps from `develop` branch.
1. Run **SignIn** app from `develop` branch.
1. If user is not created go to <http://localhost:8080/signin/createadminuser>.
1. Go to <http://localhost:8080/signin/signinuser> and log in.
1. Go to <http://localhost:8080/websiteeditor>.
1. Select "DefaultSurface" and go to "Catching rules" tab.
1. Create new catching rule for the current page, e.g. `/WebsiteEditor/surface/B/catchingrules` URI and unchecked **Nestable** field.
1. Go to "Blending points" tab and expand "Main" item.
1. Create a new pinning rule with `/signin/user` Pin URI and with the created catching rule URI (`/WebsiteEditor/surface/B/catchingrules`).
1. Go back to "Catching rules" tab and refresh the page.
1. Go back to "Blending points" tab and expand "Main" item.
1. Check the "Disabled" field and save changes.
1. Go back to "Catching rules" tab and refresh the page.

#### Expected result

`/signin/user` view does not appear on the "General" tab.

#### Actual result

`/signin/user` view still appears.

There is a strange thing in Blender. To check it you should enter into debug mode and set breakpoint on [PinningHandlers.cs:91](https://github.com/StarcounterApps/Website/blob/07049a818c8d0729259fa5295c2d29791de4e768/src/WebsiteProvider/Api/PinningHandlers.cs#L91).
After refreshing the "Catching rules" page you can retrieve the following info from Blender:

```
Blender.ListByUri("/WebsiteEditor/surface/B/catchingrules").Select(x => x.Token)
{System.Linq.Enumerable.WhereSelectListIterator<Starcounter.BlendingInfo, string>}
    Empty: "Перечисление не дало результатов"

Blender.GetRunCandidatesForUri("/WebsiteEditor/surface/B/catchingrules").Select(x => x.Uri)
{System.Linq.Enumerable.WhereSelectListIterator<Starcounter.BlendingInfo, string>}
    [0]: "/signin/user"
    [1]: "/WebsiteEditor/surface/B/catchingrules"
    [2]: "/websiteprovider/blender/surface/B/point/D/uri/BM"
```
It talks us that calling of the `/WebsiteEditor/surface/B/catchingrules` URI entails calling of the `/signin/user` URI. But also empty result of `Blender.ListByUri("/WebsiteEditor/surface/B/catchingrules").Select(x => x.Token)` tells us that `/WebsiteEditor/surface/B/catchingrules` URI is not mapped currently.

##### Starcounter version 2.3.1.7280

-----

@alemoi could you please take a look at this?

**Comment by warpech** [comment]
Is this a regression?

**Comment by warpech** [comment]
@un-tone is this a regression? Are any issues blocked by this?

@miyconst I advise this is included in 2.3.1

**Comment by warpech** [comment]
Is this the same issue as https://github.com/Starcounter/RebelsLounge/issues/195?

**Comment by un-tone** [comment]
> Is this the same issue as Starcounter/RebelsLounge#195?

These issues are different as I see. In my case the incorrectly called URI was mapped at once and unmapped then. Also I use certain tokens without contexts.

> is this a regression?

I'm not sure, possible not.

> Are any issues blocked by this?

if not fixing this the "disabling stuff" tests for Website are failing (https://github.com/StarcounterApps/Website/issues/217)

**Comment by warpech** [comment]
@un-tone could you pls check if this problem is fixed in Starcounter 2.3.1.7454?

**Comment by un-tone** [comment]
Currently there is another issue related to mapping deactivating which I'm trying to isolate.

**Comment by un-tone** [comment]
I described the new issue in #276. Please take a look.

**Comment by un-tone** [comment]
I should notice that #276 and this issue look similar but have different reasons.

**Comment by alemoi** [comment]
I tried to reproduce, but seems the steps are a bit outdated for the current Website app, because I went to look at it too late. @un-tone are you on it or otherwise I will check it.

**Comment by un-tone** [comment]
@alemoi I should check it again first. Possibly changes from #276 issue fixed this one.

**Comment by un-tone** [comment]
Yes , there is still the issue. Please check the code below:

```csharp
public static void TestBlenderUnmapping()
{
    Dictionary<String, List<BlendingInfo>> allBlendings = Blender.ListAllByUris();
    Assert.AreEqual(0, allBlendings.Count);

    String called = "";

    Handle.GET("/op1", () =>
    {
        called += "op1";
        return "op1";
    });
    Handle.GET("/op2", () =>
    {
        called += "op2";
        return "op2";
    });
    Handle.GET("/op3", () =>
    {
        called += "op3";
        return "op3";
    });

    Blender.MapUri("/op2", "token1");
    Blender.MapUri("/op1", "token1");
    Blender.MapUri("/op3", "token1");

    Response resp = Self.GET("/op3");
    Assert.AreEqual(resp.Body, "op3");
    Assert.AreEqual(called, "op3op1op2");
    called = "";

    Blender.UnmapUri("/op3", "token1");

    resp = Self.GET("/op3");
    Assert.AreEqual(resp.Body, "op3");
    Assert.AreEqual(called, "op3");
}
```

I used SC 2.3.1.7491 for testing.

**Comment by alemoi** [comment]
@un-tone do you need any help on this or maybe the issue is even fixed already?

**Comment by un-tone** [comment]
The issue is fixed. Tested in SC 2.3.1.7497.
Closing... Thanks @alemoi.
